buildSrc/src/main/java/saros/gradle/eclipse/configurator/EclipseVersionConfigurator.java

```diff
@@ -0,0 +1,50 @@
+package saros.gradle.eclipse.configurator;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.gradle.api.GradleException;
+
+/**
+ * Class that provides methods to change the eclipse plugin version defined in an osgi manifest
+ * files.
+ */
+public class EclipseVersionConfigurator {
+  private final File manifest;
+
+  public EclipseVersionConfigurator(File manifest) {
+    this.manifest = manifest;
+  }
+
+  /**
+   * Adds a qualifier to the existing version entry. (e.g. version '15.0.0' with qualifier
+   * '.NIGHTLY' becomes '15.0.0.NIGHTLY')
+   *
+   * @param qualifier Qualifier that is appended to the version
+   */
+  public void addQualifier(String qualifier) {
+    if (qualifier == null || qualifier.isEmpty()) return;
+    replaceLines("^(Bundle-Version:\\s*[0-9\\.]+)\\s*$", "$1" + qualifier);
+  }
+
+  private void replaceLines(String regex, String replacement) {
+    try {
+      Path path = manifest.toPath();
+      Charset charset = StandardCharsets.UTF_8;
+      List<String> oldContentLines = Files.readAllLines(path, charset);
+      String newContent =
+          oldContentLines
+              .stream()
+              .map(line -> line.replaceAll(regex, replacement))
+              .collect(Collectors.joining("\n"));
```

Comment:
This causes the newline at the end of the file to be dropped. A dirty fix would be to just add it back manually.

```suggestion
              .collect(Collectors.joining("\n"));

      newContent += "\n";
```